### DISCUSSION TITLE
Ribbon Tools show not used Commands

### DISCUSSION THREAD

**Discussion by rulai-yinqiang** [harborsiem/WindowsRibbon#28] [Ideas]
I add a lot of Commands, but some of them are not used in Views. I hope there will be an icon in the row to show they are not used in Views, I can consider to remove them.

**Comment by harborsiem** [comment]
Nice to have function. But I have no time for implementing.
Pull request welcome

**Comment by rulai-yinqiang** [comment]
Sorry, I'm only familiar with vb.net.

**Comment by gentleman-ming** [comment]
Hi, CommandExecuted :

Please can you add the command now, I wan the copy，paste and cut function using ctrl+c, ctrl+v, ctrl+x commands

**Reply by harborsiem** [reply]
Sorry, don't understand what you are missing.

**Reply by gentleman-ming** [reply]
I noticed that in the Ribbon, it's possible to display key tip text such as "Ctrl+N" within a button's label or tooltip. However, I'm unsure how to implement the actual keyboard shortcut (e.g., Ctrl+N) so that pressing the key combination triggers the corresponding command or action associated with the button.

Could someone please provide guidance on how to properly register or handle keyboard shortcuts (like Ctrl+N) in a Windows Ribbon application? Specifically, how to ensure that when the user presses the shortcut keys, the appropriate command is executed as if the button was clicked?

**Reply by gentleman-ming** [reply]
![test](https://github.com/user-attachments/assets/c929bea6-af76-4770-8257-00d3ef585b03)

**Reply by harborsiem** [reply]
There are 3 ways to handle shortcut keys.
2 ways are implemented in the Microsoft Windows Ribbon Framework with
1. ampersand in the LableTitle. [See](https://learn.microsoft.com/en-us/windows/win32/windowsribbon/windowsribbon-element-command-labeltitle)
2. [Keytip](https://learn.microsoft.com/en-us/windows/win32/windowsribbon/windowsribbon-element-command-keytip)
These implementations are using the ALT key, not the CTRL key.

The 3. way is an implementation in my WindowsRibbon Wrapper. You can see an example in the sample [19-Localization](https://github.com/harborsiem/WindowsRibbon/tree/master/Samples/Updated-CS/19-Localization) in the file RibbonMarkup.shortcuts. This file must be an embedded resource in your project. Here you can use also the CTRL key.
